airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt

```diff
@@ -0,0 +1,223 @@
+            is UserDefinedArray -> ArrayFieldType(recursiveArrayType(type))
+            is UserDefinedType -> PokemonFieldType
+        }
+
+    private fun recursiveArrayType(type: UserDefinedArray): JdbcFieldType<*> =
+        when (val elementType = type.elementType) {
+            is SystemType -> {
+                val leafType: JdbcFieldType<*> =
+                    leafType(elementType.typeName, elementType.scale != 0)
+                if (leafType == OffsetDateTimeFieldType) {
+                    // Mysql's JDBC driver has a bug which prevents object conversions in
+                    // ArrayDataResultSet instances. Fall back to strings instead.
+                    PokemonFieldType
+                } else {
+                    leafType
+                }
+            }
+            is UserDefinedArray -> ArrayFieldType(recursiveArrayType(elementType))
+            is UserDefinedType -> PokemonFieldType
+        }
+
+    private fun leafType(
+        typeName: String?,
+        notInteger: Boolean,
+    ): JdbcFieldType<*> =
+        // This mapping includes literals returned by the JDBC driver as well as
+        // *_TYPE_NAME column values from queries to ALL_* system tables.
+        when (typeName) {
+            "BINARY_FLOAT" -> FloatFieldType
+            "BINARY_DOUBLE" -> DoubleFieldType
+            "FLOAT",
+            "DOUBLE PRECISION",
+            "REAL", -> BigDecimalFieldType
+            "NUMBER",
+            "NUMERIC",
+            "DECIMAL",
+            "DEC", -> if (notInteger) BigDecimalFieldType else BigIntegerFieldType
+            "INTEGER",
+            "INT",
+            "SMALLINT", -> BigIntegerFieldType
+            "BOOLEAN",
+            "BOOL", -> BooleanFieldType
+            "CHAR",
+            "VARCHAR2",
+            "VARCHAR",
+            "CHARACTER",
+            "CHARACTER VARYING",
+            "CHAR VARYING", -> StringFieldType
+            "NCHAR",
+            "NVARCHAR2",
+            "NCHAR VARYING",
+            "NATIONAL CHARACTER VARYING",
+            "NATIONAL CHARACTER",
+            "NATIONAL CHAR VARYING",
+            "NATIONAL CHAR", -> NStringFieldType
+            "BLOB" -> BinaryStreamFieldType
+            "CLOB" -> ClobFieldType
+            "NCLOB" -> NClobFieldType
+            "BFILE" -> BinaryStreamFieldType
+            "DATE" -> LocalDateFieldType
+            "INTERVALDS",
+            "INTERVAL DAY TO SECOND",
+            "INTERVALYM",
+            "INTERVAL YEAR TO MONTH", -> StringFieldType
+            "JSON" -> JsonStringFieldType
+            "LONG",
+            "LONG RAW",
+            "RAW", -> BinaryStreamFieldType
+            "TIMESTAMP",
+            "TIMESTAMP WITH LOCAL TIME ZONE",
+            "TIMESTAMP WITH LOCAL TZ", -> LocalDateTimeFieldType
+            "TIMESTAMP WITH TIME ZONE",
+            "TIMESTAMP WITH TZ", -> OffsetDateTimeFieldType
+            else -> PokemonFieldType
```

Comment:
I'm guessing this works as-is because it includes the ansi sql type names but tightening this switch case to only include those which actually exist should be an action item, please file an issue to that effect.

Reply:
Thanks for pointing out. 

https://github.com/airbytehq/airbyte-internal-issues/issues/9670